invenio_records_resources/services/files/schema.py

```diff
@@ -42,23 +44,57 @@ class Meta:
         unknown = INCLUDE
 
     key = SanitizedUnicode(required=True)
+    storage_class = Str()
+    uri = Str()
+    checksum = Str()
+    size = Integer()
+
+    @validates("uri")
+    def validate_names(self, value):
+        """Validate the domain of the URI is allowed."""
+        # checking if storage class and uri are compatible is a
+        # business logic concern, not a schema concern.
+        if value:
+            domain = urlparse(value).netloc
+            allowed_domains = current_app.config.get(
+                "RECORDS_RESOURCES_FILES_ALLOWED_DOMAINS"
+            )
+            if domain not in allowed_domains:
+                raise ValidationError("Domain not allowed", field_name="uri")
+
+    @pre_dump(pass_many=False)
+    def fields_from_file_obj(self, data, **kwargs):
+        """Fields coming from the FileInstance model."""
+        # data is a FileRecord instance, might not have a file yet.
```

Comment:
Im not sure I understand fully, we are getting them form the `File` wrapper. `data.file` is a `File` object, we are dumping it into the object. We cannot use `data_key` or `attribute` since we need to load from e.g. `uri` and dump from `file.uri`.